### ISSUE TITLE
AIoHandler#exists cannot handle AFile$Default

### ISSUE THREAD

**Issue by zdenek-jonas** [microstream-one/microstream#134]
Just run the tests:
It should be ready to works without other config.
Project: Microstream-test
path: afs/oraclecloud
branch: afs

This must be something new. I have tested it for some previous version of microstream and this works. I suppose, it must have an origin in some of the latest updates.
With version from 16.7. works, after update snapshot not. 

```
java.lang.RuntimeException: Instance is not of type "interface one.microstream.afs.blobstore.BlobStoreFileWrapper" and thus cannot be handled: one.microstream.afs.AFile$Default@7cc586a8

	at one.microstream.afs.AIoHandler$Abstract.validateHandledFile(AIoHandler.java:351)
	at one.microstream.afs.AIoHandler$Abstract.exists(AIoHandler.java:429)
	at one.microstream.afs.AFile.exists(AFile.java:123)
	at one.microstream.afs.AFile$Wrapper$Abstract.exists(AFile.java:467)
	at one.microstream.afs.AFile$Wrapper$Abstract.exists(AFile.java:467)
	at one.microstream.afs.AFile$Wrapper$Abstract.exists(AFile.java:467)
	at one.microstream.afs.AbstractFileSystemTest.moveTest(AbstractFileSystemTest.java:257)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.platform.commons.util.ReflectionUtils.invokeMethod(ReflectionUtils.java:675)
	at org.junit.jupiter.engine.execution.MethodInvocation.proceed(MethodInvocation.java:60)
	at 
```
![image](https://user-images.githubusercontent.com/17253582/87790784-70da3700-c841-11ea-9293-a26e4c6c95f7.png)

**Comment by zdenek-jonas** [comment]
Not just Oracle, all the tests have failed now. For All DB

**Comment by tm-ms** [comment]
I thought I made a smart optimization last week by making the type checking for general functionality like #exists and #size stricter.
But I was only half-smart since that optimization destroyed using the general functionality for generally typed instances.
E.g. a general AFile$Default instead of a specific BlobStoreFile~.

At least I was so smart to just comment it out instead of deleting it, so it was quickly reverted.
Like three-quarter smart or so.

I also added a comment to prevent a short-sighted fatal optimization from happening again.
